hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/SCMCommonPlacementPolicy.java

```diff
@@ -264,6 +264,16 @@ public static boolean hasEnoughSpace(DatanodeDetails datanodeDetails,
   public abstract DatanodeDetails chooseNode(
       List<DatanodeDetails> healthyNodes);
 
+  /**
+   * Choose a datanode according to the policy, this function is implemented
+   * by the actual policy class.
+   *
+   * @param healthyNodes - Set of healthy nodes we can choose from.
+   * @return DatanodeDetails
+   */
+  public abstract DatanodeDetails chooseFirstNode(
```

Comment:
I don't think it's necessary to define this in the parent class, since it looks like all the container based implementations are not used. I think just defining `chooseFirstNode` as `private` in `PipelinePlacementPolicy` would suffice, right?

Reply:
Oh right. I was hesitating between these two when I made changes before.